packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1839,9 +1841,6 @@ function prepareFreshStack(root: FiberRoot, lanes: Lanes): Fiber {
     }
   }
 
-  root.finishedWork = null;
-  root.finishedLanes = NoLanes;
```

Comment:
Should `root.cancelPendingCommit` and `timeoutHandle` get reset?

Reply:
They do get reset right below here as they're cancelled.